app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,28 @@
+class RentalsController < ApplicationController
+  
+  def checkout 
+    rental = Rental.new(customer_id: params[:rental][:customer_id], movie_id: params[:rental][:movie_id], check_out_date: Date.today, due_date: (Date.today) + 7)
+        
+    if rental.save
+      render json: rental.as_json(only: [:customer_id, :movie_id]), status: :ok
+      return
+    else
+      # binding.pry
+      render json: { ok: false, "errors" => ["Not Found"]}, status: :not_found
+      return
+    end
+  end
+
+  def checkin 
```

Comment:
Checkin should not create a new rental as you should be checking in an existing rental. You need to first find the rental and then checkin. Some of this logic would go in a custom model method.

Reply:
In addition, checkin and checkout should affect available movie inventory -- this will likely be a custom method in the movie model.

Reply:
I realize now that movie inventory is optional -- I talked about this with Cloudy